languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py

```diff
@@ -53,46 +53,48 @@ def and_filter(current, new):
 def partial_to_filter(expression: Expression, session: Session, model, get_model):
     """Convert constraints in ``partial`` to a filter over ``model`` that should be applied to query."""
     expression = preprocess(expression)
+    roles_method = check_for_roles_method(expression)
 
-    role_allows = False
-    if contains_role_allows(expression):
-        role_allows = True
+    return (
+        translate_expr(expression, session, model, get_model),
+        roles_method,
+    )
 
-    return (translate_expr(expression, session, model, get_model), role_allows)
 
-
-def contains_role_allows(expression: Expression):
-    def _is_role_allow(op, left, right):
-        op = expr.operator
-        left = expr.args[0]
-        right = expr.args[1]
-        is_role_allow = (
+def check_for_roles_method(expression: Expression):
+    def _is_roles_method(op, left, right):
+        is_roles_method = (
             left is True
```

Comment:
So the expression comes in from the core as `(true).user_in_role()`?

Reply:
It's actually `True = OsoRoles.user_in_role()`

Reply:
The expression here is a `Unify` op